### ISSUE TITLE
Uncaught TypeError: data.indexOf is not a function in _handle_msg_from_outside method in host.js

### ISSUE THREAD

**Issue by joeyaza** [InteractiveAdvertisingBureau/safeframe#21]
I get the above error as it expects `data` to be a string but in my case is coming in as an object from the PostMessage. And running .indexOf on a non-string throws an error.
Is there any way to counter this?
potentially, data could be stringified so its certain to be a string? or typeof data could be ascertained prior to make sure it is a string before running the indexOf method..

**Comment by seansd-zz** [comment]
Yeah the code was written before automatic JSON parsing of objects sent via postMessage started happening.  Since the communication protocol is always expected to be a string internally I would do a typeof check and then reject the message if it's not a string.

**Comment by joeyaza** [comment]
shall I change and make a PR ?

**Comment by seansd-zz** [comment]
sure go for it

**Comment by kevingilbert100** [comment]
Any update here?

**Comment by codexandr** [comment]
I have same issue. how can it be solve?
